### ISSUE TITLE
Usage of SO_REUSEPORT

### ISSUE THREAD

**Issue by hinxx** [sammyfreg/netImgui#41]
During my tests on Linux I noticed that I could run two instanced of the server and both would listen on the same port (i.e. default 8888). After that the clients (SampleNoBackend in my case) would connect to one or the other instance of the server in an arbitrary manner.

I think this behavior is due to the use of `SO_REUSEPORT` found here

https://github.com/sammyfreg/netImgui/blob/c7071017c5bb04727ded1b62be1dec7fcfef95e4/Code/Client/Private/NetImgui_NetworkPosix.cpp#L86

I guess this is useful if the server(s) are making the connections to the client(s) but in the opposite direction it results in undefined behavior; we never know which server is going to accept the client's connection first.

I would expect that the client is capable of specifying which server port it wants to connect to (with existing default port) and the servers would open their own listening port (again with existing default). In this scenario the servers should fail to open the same listening port more than once.

**Comment by sammyfreg** [comment]
This was added to allow Server to start listening to port, even if port wasn't properly closed from the previous instance.

Ideally, I should support 2 running instances of the Server Application, but prevent listening on the same port...

**Comment by sammyfreg** [comment]
But maybe I could do a connection test first, before opening the Server connection.

**Comment by hinxx** [comment]
> This was added to allow Server to start listening to port, even if port wasn't properly closed from the previous instance.

I tried commenting out the `SO_REUSEPORT` line, would start the server and then went and killed it with different signals using `kill -x PID`, where x was 2, 9, 15. When trying to restart the server seconds after it always managed to listen to the same port.

FWIW, I have observed in the past (not with your server code) that if the process is not cleanly terminated upon restart it could fail to claim the the same IP address ; you are already solving that with the `SO_REUSEADDR` socket option. I've been now trying to get that error with your server (after commenting out the `SO_REUSEADDR` line) but I can not seem to get it..

**Comment by sammyfreg** [comment]
I tried killing the Server application on Windows and just like on linux, there doesn't seem to be any issues.

So, with bb4fcda71bdbb030223b603308d89de3afd6e975 submited, I made this behaviour optional, and disabled by default.

So now, when opening 2 instances of the NetImguiServer, the first one can open the port, and the second one will try opening it every 500ms, so it can receive it once the 1st application is closed. 

Note: If people really want 2 NetImguiServers running, then can put each in a different directory, and give them a different listening port in their config.
